packages/venia-concept/src/components/SearchBar/searchBar.js

```diff
@@ -0,0 +1,128 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { compose } from 'redux';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+
+import classify from 'src/classify';
+import { executeSearch } from 'src/actions/app';
+import defaultClasses from './searchBar.css';
+
+import Icon from 'src/components/Icon';
+
+export class SearchBar extends Component {
+    static propTypes = {
+        classes: PropTypes.shape({
+            root: PropTypes.string,
+            searchBlock: PropTypes.string,
+            searchBlockOpen: PropTypes.string,
+            searchBar: PropTypes.string,
+            searchIcon: PropTypes.string
+        }),
+        isOpen: PropTypes.bool,
+        executeSearch: PropTypes.func
+    };
+
+    constructor(props) {
+        super(props);
+        this.searchRef = React.createRef();
+        this.clearRef = React.createRef();
+    }
+
+    async componentDidMount() {
+        if (this.props.isOpen) {
+            this.searchRef.current.focus();
+        }
+        if (document.location.pathname === '/search.html') {
+            const params = new URL(document.location).searchParams;
+            this.searchRef.current.value = params.get('query');
+            this.setClearIcon(this.searchRef.current.value);
+        }
```

Comment:
This is currently being addressed by wrapping the component with "withRouter", and then getting location using the route props instead of pulling from document.location. However, we may want to limit our use of additional wrappers and use a Route component as suggested.